test_scripts/pypeit_test.py

```diff
@@ -0,0 +1,1159 @@
+            setup.tests.append(PypeItReduceTest(setup, pargs.masters, std=True))
+
+        # Try re-running keck_lris_red setups with masters
+        if setup.instr == 'keck_lris_red':
+            setup.tests.append(PypeItReduceTest(setup, True))
+
+    if pargs.tests == 'reduce':
+        # Skip the afterburner tests
+        return setup
+    # ----------------------------------------------------------
+
+    if not flg_ql:
+        # ----------------------------------------------------------
+        # SensFunc tests
+        # ----------------------------------------------------------
+        # UVIS algorithm without .sens file
+        if setup.instr == 'shane_kast_blue' and '600' in setup.name:
+            # TODO: Instead check for file with standard star in the
+            # file name?
+            std_file = 'spec1d_b24-Feige66_KASTb_2015May20T041246.960.fits'
+            setup.tests.append(PypeItSensFuncTest(setup, std_file=std_file))
+
+        # IR algorithm with .sens file
+        if setup.instr == 'gemini_gnirs' and setup.name == '32_SB_SXD':
+            std_file = 'spec1d_cN20170331S0206-HIP62745_GNIRS_2017Mar31T083351.681.fits'
+            sens_file = os.path.join(setup.dev_path, 'sensfunc_files', 'gemini_gnirs_32_sb_sxd.sens')
+            setup.tests.append(PypeItSensFuncTest(setup, std_file=std_file, sens_file=sens_file))
+
+        # IR algorithm, multi-slit, without .sens file
+        if setup.instr == 'gemini_gmos' and setup.name == 'GS_HAM_R400_860':
+            std_file = 'spec1d_S20181219S0316-GD71_GMOS-S_1864May27T230832.356.fits'
+            setup.tests.append(PypeItSensFuncTest(setup, std_file=std_file))
+
+        # ----------------------------------------------------------
+
+        # ----------------------------------------------------------
+        # fluxing/coadd1d/tellfit tests
+        # ----------------------------------------------------------
+        flux_coadd1d_pairs = {'shane_kast_blue': '600_4310_d55',  # long-slit
+                              'gemini_gnirs': '32_SB_SXD',  # echelle
+                              'gemini_gmos': 'GS_HAM_R400_860'}  # multi-slit
+
+        if any([setup.instr == key and setup.name == flux_coadd1d_pairs[key]
+                for key in flux_coadd1d_pairs.keys()]):
+            setup.tests.append(PypeItFluxSetupTest(setup))
+            setup.tests.append(PypeItFluxTest(setup))
+            setup.tests.append(PypeItCoadd1DTest(setup))
+
+        # ----------------------------------------------------------
+
+        # ----------------------------------------------------------
+        # 2D coadding tests
+        # ----------------------------------------------------------
+        # Echelle
+        if setup.instr == 'gemini_gnirs' and setup.name == '32_SB_SXD':
+            setup.tests.append(PypeItCoadd2DTest(setup, obj='pisco'))
+
+        # MultiSlit
+        if setup.instr == 'keck_lris_blue' and setup.name == 'multi_600_4000_d560':
```

Comment:
I'd also like to remove these `if` statements and just
have `coadd2d` run on a list defined up above.